s3repo/model.py

```diff
@@ -81,22 +104,28 @@ def __init__(self, s3_settings):
         self.sync_thread.start()
 
     @staticmethod
-    def _format_paths(dist_path, dist_version, dist_base, filename, product):
+    def _format_paths(dist_path: str, dist_version: str, dist_base: str,
+                      filename: str, product: str) -> List[Tuple[str, str]]:
         """Formats the file path and repository path according
         to the filename and distribution information.
         Returns an array of tuples (repo_path, path).
         """
-        result = []
-        file_type_err = 'The "{0}" file does not match the type of files ' +\
-            'used in the {1}-based repositories.'
+        result: List[Tuple[str, str]] = []
+        file_type_err = 'The "{0}" file does not match the type of files ' + \
+                        'used in the {1}-based repositories.'
         if dist_base == 'rpm':
             if re.fullmatch(r'.*\.(x86_64|aarch64)\.rpm', filename):
                 # Example of the path for x86_64 rpm repository:
                 # .../live/1.10/fedora/31/x86_64
+                arch_filename = re.match(
+                    r'.*\.(?P<arch>(x86_64|aarch64))\.rpm', filename)
+                if arch_filename is None:
+                    raise Exception(
+                        "Either wrong name, or arch is not supported.")
                 repo_path = '/'.join([
                     dist_path,
                     dist_version,
-                    re.match(r'.*\.(?P<arch>(x86_64|aarch64))\.rpm', filename).group('arch')
+                    arch_filename.group('arch')
```

Comment:
Not related with the commit.
What's the code style rule being violated here?

Reply:
rolled back, but it brings mypy warning